lexibank_kitchensemitic.py

```diff
@@ -1,187 +1,110 @@
-                    if len(forms) > len(cogs):
-                        cogs.extend('-' for _ in range(len(forms) - len(cogs)))
-                    
-                    # there does appear to be one instance of more cognates than forms:
-                    #   Harsusi's "ges'" - RAIN (N.) = D/I
-                    # .. but it looks cognate with Hebrew's and Ugaritic's set D rather than
-                    # set I (e.g. Ogaden Arabic "matar"), so I think it's a typo and
-                    # we'll ignore it.
-                    
-                    # there are also forms that have no lexeme but do have a cognate set...
-                    # have to ignore these for now. Can't think of what to do with them
-                    # otherwise:
-                    #    Mɛhri Dog --- D
-                    #    Mɛhri Dry (adj.) --- F
-                    #    Hebrew Grass --- C/J
-                    #    Hebrew New --- A
-                    #    Mɛhri Seed --- C
-                    #    Hebrew Three --- A
-                    #    Mɛhri Warm --- H
-                    #    Hebrew Ye --- A
-
-                    for form, cog in zip(forms, cogs):
-                        assert len(form), "Empty form %s - %r - %r - %r" % (lid, gloss, form, cog)
-                        assert len(cog), "Empty cognate %s - %r - %r - %r" % (lid, gloss, form, cog)
-                        
-                        cogid = '%s-%s' % (csid, cog)
-                        for row in ds.add_lexemes(
-                                Language_ID=lid,
-                                Parameter_ID=csid,
-                                Value=form,
-                                Source=src
-                            ):
-                            if cog != '-':
-                                ds.add_cognate(lexeme=row, Cognateset_ID=cogid, Source=['Kitchen2009'])
+    language_class = KSLanguage
+
+    form_spec = FormSpec(
+        brackets={}, separators="/,", missing_data=("---",), strip_inside_brackets=False
+    )
+
+    def cmd_download(self, args):
+        self.raw_dir.xls2csv("Semitic.Wordlists.xls")
+        self.raw_dir.xls2csv("Semitic.Codings.Multistate.xlsx")
+
+    def cmd_makecldf(self, args):
+        lexeme_rows = self.raw_dir.read_csv("Semitic.Wordlists.ActualWordlists.csv")
+        lexeme_header = lexeme_rows.pop(0)
+
+        cognate_table = make_cognate_table(
+            self.raw_dir.read_csv("Semitic.Codings.Multistate.Sheet1.csv")
+        )
+
+        languages, concepts = {}, {}
+
+        for concept in self.conceptlist.concepts.values():
+            concepts[concept.english] = concept.id
+
+        args.writer.add_concepts(id_factory=lambda c: c.id)
+
+        for language in self.languages:
+            languages[language["Name"]] = (language["ID"], language["Sources"].split(";"))
+
+        args.writer.add_languages()
+
+        for row in pb(lexeme_rows, desc=f"Build CLDF for {self.id}"):
```

Comment:
Is it probably useful to give a default for the `pb` function, like `building cldf`, to save code, as we always use the `desc=` construct so far?

Reply:
This is no critic, but a question to all lexibankers.

Reply:
Sounds good!

Reply:
If we keep `pb` - which is really just a tiny wrapper around `tqdm.tqdm` - I'd advocate renaming it to `pylexibank.progress` - and also making it importable from `pylexibank`. Generally, I think it would be good for readability if **everything** public in `pylexibank` was exposed on the top-level, and imported and used with full name, e.g.:
```python
import pylexibank


class Dataset(pylexibank.Dataset):
...
    def cmd_makecldf(...):
        for x in pylexibank.progress(...):
            ...

Reply:
Agree. I'm happy to live without progress bars however.

Reply:
Yes, I agree as well.